src/sagemaker/jumpstart/curated_hub/hub_model_specs/hub_model_specs.py

```diff
@@ -0,0 +1,241 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module data structures for Private Hub content."""
+from __future__ import absolute_import
+
+from dataclasses import dataclass
+from typing import Optional, List, Mapping
+from enum import Enum
+from sagemaker.jumpstart.types import JumpStartHyperparameter
+
+
+class DependencyType(str, Enum):
+    """Type of dependency for Private Hub model"""
+
+    SCRIPT = "Script"
+    ARTIFACT = "Artifact"
+    DATASET = "Dataset"
+    NOTEBOOK = "Notebook"
+    OTHER = "Other"
+
+
+@dataclass
```

Comment:
this is a cool construct, definitely makes everything cleaner

Reply:
ty